web/web.go

```diff
@@ -84,34 +84,71 @@ func openWebBrowser(url string) error {
 	return exec.Command(cmd, args...).Start()
 }
 
+func startGrpcWebProxy(ctx *progctx.ProgCtx) error {
+	grpcWebProxyProc = exec.CommandContext(ctx, "grpcwebproxy", []string{
+		fmt.Sprintf("--backend_addr=localhost:%d", grpcWebProxyParams.grpcServicePort),
+		"--run_tls_server=false",
+		"--allow_all_origins",
+		"--server_http_max_read_timeout=1h",
+		"--server_http_max_write_timeout=1h",
+		fmt.Sprintf("--server_bind_address=%s", grpcWebProxyParams.serverBindAddress),
+		fmt.Sprintf("--server_http_debug_port=%d", grpcWebProxyParams.serverHttpDebugPort),
+	}...)
+
+	return grpcWebProxyProc.Start()
+}
+
 func assureGrpcWebProxyRunning(ctx *progctx.ProgCtx) error {
 	if grpcWebProxyProc == nil {
-		proc := exec.CommandContext(ctx, "grpcwebproxy", []string{
-			fmt.Sprintf("--backend_addr=localhost:%d", grpcWebProxyParams.grpcServicePort),
-			"--run_tls_server=false",
-			"--allow_all_origins",
-			"--server_http_max_read_timeout=1h",
-			"--server_http_max_write_timeout=1h",
-			fmt.Sprintf("--server_bind_address=%s", grpcWebProxyParams.serverBindAddress),
-			fmt.Sprintf("--server_http_debug_port=%d", grpcWebProxyParams.serverHttpDebugPort),
-		}...)
-
-		if err := proc.Start(); err != nil {
+		if err := startGrpcWebProxy(ctx); err != nil {
 			return err
 		}
 
-		grpcWebProxyProc = proc
+		simplelogger.Infof("grpcwebproxy started.")
+
 		ctx.WaitAdd("grpcwebproxy", 1)
 		go func() {
 			defer ctx.WaitDone("grpcwebproxy")
 
 			err := grpcWebProxyProc.Wait()
 			if err != nil && ctx.Err() == nil {
-				simplelogger.Errorf("grpcwebproxy exit unexpectedly: %v", err)
+				simplelogger.Warnf("grpcwebproxy exit unexpectedly: %v, try restarting ...", err)
+
+				tryKillExistingGrpcWebProxyProcess()
+
+				if err = startGrpcWebProxy(ctx); err != nil {
```

Comment:
Looks like we should put `tryKillExistingGrpcWebProxyProcess` inside `startGrpcWebProxy`.

Reply:
Why? Do you mean we should always kill before start? If so I intentionally didn't do that because killing grpcwebproxy causes about 100ms delay. I don't want to have this delay when the new grpcwebproxy is executing well. 
thoughts?

Reply:
More comments on this? @wgtdkp

Reply:
No strong preference, both work for me.